src/GraphQL/Resolvers/Resolver.php

```diff
@@ -0,0 +1,201 @@
+        }
+
+        if (!$elementalArea->canEdit($context[QueryHandler::CURRENT_USER])) {
+            throw new InvalidArgumentException("The current user has insufficient permission to edit ElementalAreas");
+        }
+
+        /** @var BaseElement $newElement */
+        $newElement = Injector::inst()->create($elementClass);
+
+        if (!$newElement->canEdit($context[QueryHandler::CURRENT_USER])) {
+            throw new InvalidArgumentException(
+                'The current user has insufficient permission to edit Elements'
+            );
+        }
+
+        // Assign the parent ID directly rather than via HasManyList to prevent multiple writes.
+        // See BaseElement::$has_one for the "Parent" naming.
+        $newElement->ParentID = $elementalArea->ID;
+        // Ensure that a sort order is assigned - see BaseElement::onBeforeWrite()
+        $newElement->onBeforeWrite();
+
+        if ($afterElementID !== null) {
+            /** @var ReorderElements $reorderer */
+            $reorderer = Injector::inst()->create(ReorderElements::class, $newElement);
+            $reorderer->reorder($afterElementID); // also writes the element
+        } else {
+            $newElement->write();
+        }
+
+        return $newElement;
+    }
+
+    /**
+     * @param $object
+     * @param array $args
+     * @param $context
+     * @param ResolveInfo $info
+     * @return BaseElement
+     * @throws ValidationException
+     */
+    public static function resolveDuplicateBlock($object, array $args, $context, ResolveInfo $info)
+    {
+        // load element to clone
+        $elementID = $args['id'];
+        $element = BaseElement::get_by_id($elementID);
+        if (!$element) {
+            throw new InvalidArgumentException("Invalid BaseElementID: $elementID");
+        }
+
+        // check can edit the elemental area
+        $areaID = $element->ParentID;
+        $area = ElementalArea::get_by_id($areaID);
+        if (!$area) {
+            throw new InvalidArgumentException("Invalid ParentID on BaseElement: $elementID");
+        }
+        if (!$area->canEdit($context[QueryHandler::CURRENT_USER])) {
+            throw new InvalidArgumentException(
+                "The current user has insufficient permission to edit ElementalArea: $areaID"
+            );
+        }
+
+        try {
+            // clone element
+            $clone = $element->duplicate(false);
+            $clone->Title = static::newTitle($clone->Title);
+            $clone->Sort = 0; // must be zeroed for reorder to work
+            $area->Elements()->add($clone);
+
+            // reorder
+            $reorderer = Injector::inst()->create(ReorderElements::class, $clone);
+            $reorderer->reorder($elementID);
+
+            return $clone;
+        } catch (Exception $e) {
+            throw new Exception("Something went wrong when duplicating element: $elementID");
+        }
+    }
+
+
+    public static function newTitle(string $title): string
```

Comment:
Title is optional (can be blank in new blocks) - regression from GraphQL v3 functionality.

Reply:
Are you saying the title should be nullable? I'm looking at this: https://github.com/dnadesign/silverstripe-elemental/blob/4/src/GraphQL/DuplicateElementMutation.php#L78

Reply:
Needed null coalescing, see bfa278044ef9ef9fe13ef1cc82c2c7e1bc1ad95e